ksql-engine/src/main/java/io/confluent/ksql/function/Blacklist.java

```diff
@@ -26,31 +26,30 @@
 import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 /**
  * Used to restrict the classes that can be loaded by user supplied UDFs
  */
 public class Blacklist implements Predicate<String> {
   private static final Logger logger = LoggerFactory.getLogger(Blacklist.class);
-  private static final String EMPTY_BLACKLIST = "^(?)\\.?.*$";
+  private static final String EMPTY_BLACKLIST = "^(?:)\\.?.*$";
 
-  private String blackList;
+  private String blackList = ".*";
 
   Blacklist(final File inputFile) {
     try {
-      final StringBuilder builder = new StringBuilder("^(?:");
-      Files.readLines(inputFile, Charset.forName(StandardCharsets.UTF_8.name()))
-          .forEach(item -> {
-            final String trimmed = item.trim();
-            if (!(trimmed.isEmpty() || trimmed.startsWith("#"))) {
-              builder.append(trimmed.replaceAll("\\.", "\\\\.")).append("|");
-            }
-          });
-      builder.deleteCharAt(builder.length() - 1);
-      builder.append(")\\.?.*$");
-      this.blackList = builder.toString().equals(EMPTY_BLACKLIST)
-          ? ""
-          : builder.toString();
+      this.blackList = Files.readLines(inputFile, Charset.forName(StandardCharsets.UTF_8.name()))
```

Comment:
It would be nice to have a comment describing how this block is parsing the blacklist file. Specifically, how are strings being transformed with replace and join? The regexes are not straightforward to interpret.